pkg/webhook/ingress/validating.go

```diff
@@ -155,6 +164,35 @@ func (r *handler) validateIngress(ctx context.Context, c client.Client, object I
 		return admission.Errored(http.StatusBadRequest, NewIngressClassForbidden(*ingressClass, *tnt.Spec.IngressClasses))
 	}
 
+	//TODO extract logic below into a method
+	if tnt.Spec.IngressHostnames == nil {
+		return admission.Allowed("")
+	}
+
+	valid = false
+	matched = false
+	hostnames := ingress.Hostnames()
+	if len(hostnames) > 0 {
+		valid = tnt.Spec.IngressHostnames.Allowed.AreStringsInList(hostnames)
```

Comment:
Here I verify if all strings are in the allowed list.

Reply:
LGTM, although  a different approach to what we implemented with other components such as the Container Registry check

https://github.com/clastix/capsule/blob/98e441f1e9a9e24630624fcb8c68ff5039b59e28/pkg/webhook/registry/validating.go#L85-L94

It's just a smell of code stuff, maybe iterating over the array would make it easier to understand the business logic behind the webhook.

Let's start the discussion.

Reply:
My idea here would be to have a general purpose matching between two arrays. I need to verify that all elements in the "hostnames" array are contained in the allowed list.

Reply:
I thought about this and checking a list of strings against doesn't return the missed intersection, or rather the set of strings not in the defined list.

Since in the end we're iterating over using `range`, let's avoid to use a func and implement this loop here so we can gather the item not in the array.

Reply:
fine for me.